airflow/models/trigger.py

```diff
@@ -244,8 +245,9 @@ def assign_unassigned(
     def get_sorted_triggers(cls, capacity, alive_triggerer_ids, session):
         query = with_row_locks(
             select(cls.id)
+            .join(TaskInstance, cls.id == TaskInstance.trigger_id, isouter=False)
```

Comment:
Since there is no Triggers to process without being associated to a TI (if there is no TI with the trigger id, it means that the trigger is already executed and has returned an event) , it should be fine to use inner join here.